crates/core/plugin_sm/src/plugin.rs

```diff
@@ -461,13 +468,35 @@ impl Plugin for ExternalPluginCommand {
         let command = self.command(LIST, None)?;
         let output = self.execute(command, logger).await?;
         if output.status.success() {
+            let filtered_output = match (&self.exclude, &self.include) {
+                (None, None) => output.stdout,
+                _ => {
+                    let exclude_filter =
+                        Regex::new(self.exclude.as_ref().unwrap_or(&r".*".to_string()))?;
+                    let include_filter =
+                        Regex::new(self.include.as_ref().unwrap_or(&r"^$".to_string()))?;
```

Comment:
That's correct, but deserves comments: this PR got too many comments/questions on that point.

```suggestion
                    // If no exclude pattern is given, exclude everything (except what matches the include pattern)
                    let exclude_filter =
                        Regex::new(self.exclude.as_ref().unwrap_or(&r".*".to_string()))?;
                    // If no include pattern is given, include nothing (except what doesn't match the exclude pattern)   
                    let include_filter =
                        Regex::new(self.include.as_ref().unwrap_or(&r"^$".to_string()))?;
```